utils.py

```diff
@@ -926,4 +929,103 @@ def quote_field_name(f):
     return '`{}`'.format(f) if re.search('^\d|\.', f) else f
 
 
+def merge_TStructs(s):
+    """
+
+    Merges multiple TStructs together and outputs a new TStruct with the union of the fields.
+    Notes:
+    - If a leaf field is found in multiple input TStructs, its type has to be the same
+    - Attributes are merged. If the same attribute is found in different TStructs it should have the same value.
+
+    :param list s: List of Structs to merge
+    :return: Merged Struct
+    :rtype: TStruct
+
+    """
+
+    if len(s) < 2:
+        return s.pop()
+
+    fields = OrderedDict()
+    s_fields = [flatten_struct(x, root='', recursive=False) for x in s]
+
+    while len(s_fields) > 0:
+        s_current = s_fields.pop(0)
+        for name, f in s_current.iteritems():
+            if name not in fields:
+                attributes = f.attributes
+                f_overlap = [x[name] for x in s_fields if name in x]
+
+                for f2 in f_overlap:
+                    if not isinstance(f2.typ, type(f.typ)):
+                        sys.exit("ERROR: Cannot merge structs with type {} and {}".format(f.typ, f2.typ))
+                    for k,v in f2.attributes.iteritems():
+                        if k in attributes:
+                            if v != attributes[k]:
+                                logger.warn("Found different values for attribute {} for field {} while merging structs:{}, {}".format(k,name,attributes[k],v))
+                        else:
+                            attributes[k] = v
+
+                if isinstance(f.typ, TStruct) and f_overlap:
+                    fields[name] = Field(name, merge_TStructs([f.typ] + [f2.typ for f2 in f_overlap]))
+                else:
+                    fields[name] = f
+
+                fields[name].attributes = attributes
+
+    return TStruct.from_fields(fields.values())
```

Comment:
Is the `OrderedDict` just to avoid having to check the name of everything in fields? Sounds good if so, just curious.

Reply:
Yeah, it's for O(1) lookups -- I could use a list too as it really wouldn't make a big difference.